### PR TITLE
[Ng1] rlSelect, close multiple menus

### PR DESCRIPTION
**Pull request by BrentWMiller**
**Youtrack issue: https://renovo.myjetbrains.com/youtrack/issue/THM-74**

The original issue was that when you opened an rlSelect menu and then another rlSelect menu the first wouldn't close. So the menus would start to stack, or layer on top of one another.

This switches the ng1 select from using `ng-click` to using `ng-mousedown` and `ng-blur`. It also drops the need for the `rl-off-click` behavior.

These commits are a lot of trial and error. The last 4 commits are the final idea, once I figured out that `ng-focus` was breaking it and I could switch back to a toggle/close.

ng1-select-fix

### CODE REVIEW

File: source/components/inputs/select/select.ng1.ts [hunk 1]
```diff
@@ -96,10 +96,8 @@ export class SelectController extends InputController {
 		this.showOptions = !this.showOptions;
 	}
 
-	close: { (): void } = () => {
-		if (this.showOptions) {
```
Comment (KorinaGraber) [inline source/components/inputs/select/select.ng1.ts:100]: Why this change? If showOptions is already true, it shouldn't matter whether you clear it or not.

**Review by KorinaGraber — CHANGES_REQUESTED** [review]
Question

File: source/components/inputs/select/select.ng1.tests.ts [hunk 2]
```diff
@@ -117,14 +117,9 @@ describe('SelectController', () => {
...
 			dropdown.showOptions = true;
 			dropdown.close();
-			expect(dropdown.showOptions).to.be.false;
-		});
 
-		it('should do nothing if the options are already closed', (): void => {
-			dropdown.showOptions = false;
-			dropdown.close();
 			expect(dropdown.showOptions).to.be.false;
-		});
```
Comment (KorinaGraber) [inline source/components/inputs/select/select.ng1.tests.ts:127]: If the below changes, this test can come back too

File: source/components/inputs/select/select.ng1.ts (hunk 1 above)
Comment (KorinaGraber) [reply source/components/inputs/select/select.ng1.ts:100]: It looks like this more or less just happened because you were trying different things so when you came back to this it was different than when you'd started...?

Comment (BrentWMiller) [reply source/components/inputs/select/select.ng1.ts:100]: It was lost while trying different things and the newer method  worked how I wanted it. Is there an advantage to the original method versus mine?

Comment (KorinaGraber) [reply source/components/inputs/select/select.ng1.ts:100]: I was thinking about that after I commented. I can't really think of a reason, so we can probably leave it as-is.

**Review by KorinaGraber — APPROVED** [review]

File: source/components/inputs/select/select.ng1.ts (hunk 1 above)
Comment (BrentWMiller) [reply source/components/inputs/select/select.ng1.ts:100]: 👍 Okay, I just wanted to make sure I wasn't misunderstanding something.
